File: 0001-TwoSum.py
from typing import List

class Solution:
    """Method to find the indices of numbers that add up to the target"""
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        opposite = {}
        for i, num in enumerate(nums):
            if num in opposite:
                return [i, opposite[num]]
            opposite[target - num] = i


S = Solution()
print(S.twoSum([3, 2, 4], 6))

File: 0001-TwoSum.ts
/* 
Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
You may assume that each input would have exactly one solution, and you may not use the same element twice.
You can return the answer in any order.
*/

function twoSum(nums: number[], target: number): number[] | undefined {
  const seenNums = {} as Record<number, number>;
  for (let i = 0; i < nums.length; i += 1) {
    if (seenNums.hasOwnProperty(nums[i])) {
      return [seenNums[nums[i]], i];
    }
    seenNums[target - nums[i]] = i;
  }
};

console.log(twoSum([2, 7, 11, 15], 9));